compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/diagnostic/BLangDiagnosticLog.java

```diff
@@ -49,13 +47,15 @@
 
     private int errorCount = 0;
     private PackageCache packageCache;
+    private PackageID currentPackageId;
```

Comment:
Having the current package ID in the logger can be problematic. Only one logger is created per compilation. But a single compilation can have multiple modules.

So here it is expected to set the `currentPackageId` before visiting the package. That makes us depend on the order of usage of the methods, and also makes it easy for someone to miss it out.

Shall we send package ID as a parameter to the diagnostic reporting function every-time?

Reply:
Will do this change in another PR